Salwador/Views/ImageGenerationView.swift

```diff
@@ -88,29 +85,29 @@ private extension ContentView {
 
     var imageView: some View {
         VStack {
-            if hasError {
+            if viewModel.hasError {
```

Comment:
suggestion: here we could react to a enum `@State` variable on response from API. 

Here is some follow up on this https://www.hackingwithswift.com/quick-start/swiftui/what-is-the-state-property-wrapper

Reply:
Good point, all the comments about enum I will try to implement in the new PR.